src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -714,22 +719,64 @@ class ResourceService(
   def loadPolicy(policyId: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Option[AccessPolicy]] =
     accessPolicyDAO.loadPolicy(policyId, samRequestContext)
 
-  def loadResourcePolicy(policyIdentity: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Option[AccessPolicyMembership]] =
+  def loadResourcePolicy(policyIdentity: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Option[AccessPolicyMembershipResponse]] =
     accessPolicyDAO.loadPolicyMembership(policyIdentity, samRequestContext)
 
   private def makeValidatablePolicies(
-      policies: Map[AccessPolicyName, AccessPolicyMembership],
+      policies: Map[AccessPolicyName, AccessPolicyMembershipRequest],
       samRequestContext: SamRequestContext
   ): IO[Set[ValidatableAccessPolicy]] =
     policies.toList
-      .traverse { case (accessPolicyName, accessPolicyMembership) =>
-        makeCreatablePolicy(accessPolicyName, accessPolicyMembership, samRequestContext)
+      .traverse { case (accessPolicyName, accessPolicyMembershipRequest) =>
+        // grouping member emails and policy emails together
+        val allEmails = getPolicyEmailsFromAccessPolicyMembership(accessPolicyMembershipRequest, samRequestContext)
+          .map { ioPolicyEmails =>
+            for {
+              policyEmails <- ioPolicyEmails
+            } yield accessPolicyMembershipRequest.memberEmails ++ policyEmails
+          }
+          .getOrElse(IO(accessPolicyMembershipRequest.memberEmails))
```

Comment:
Nit: it feels like allEmails could be in the for comp below with the other stuff but not 100% sure